src/schema/v2/previewSavedSearch.ts

```diff
@@ -102,30 +103,45 @@ const PreviewSavedSearchType = new GraphQLObjectType<any, ResolverContext>({
       type: new GraphQLNonNull(
         new GraphQLList(new GraphQLNonNull(SearchCriteriaLabel))
       ),
-      resolve: async (_root, _args, { filterArtworksLoader }) => {
-        const suggestedFiltersByArtist: SearchCriteriaLabel[][] = await Promise.all(
-          _root.artistIDs.map((artistSlug) =>
-            getSuggestedFiltersByArtistSlug(artistSlug, filterArtworksLoader)
-          )
+      resolve: async ({ artistIDs }, _args, { filterArtworksLoader }) => {
+        if (!artistIDs) {
+          throw new Error("artistIDs are required to get suggested filters")
+        }
+
+        const gravityArgs = {
+          size: 0,
+          published: true,
+          aggregations: ["attribution_class", "medium"],
+          artist_ids: artistIDs,
+        }
+
+        const { aggregations } = await filterArtworksLoader(gravityArgs)
+
+        const suggestedFilters: SearchCriteriaLabel[] = []
+
+        const rarity = getRaritySearchCriteriaLabel(
+          getMostPopularOption(aggregations["attribution_class"]).value
         )
 
-        return (
-          _.chain(suggestedFiltersByArtist)
-            .flatten()
-            // Remove duplicates
-            .uniqBy((searchCriteria) => JSON.stringify(searchCriteria))
-            // Group by search criteria type
-            .groupBy((searchCriteria) => searchCriteria.field)
-            .values()
-            .flatten()
-            .value()
+        if (rarity && SUPPORTED_ATTRIBUTIONS_CLASSES.includes(rarity.value)) {
+          suggestedFilters.push(rarity)
+        }
+
+        const medium = getMediumSearchCriteriaLabel(
+          getMostPopularOption(aggregations["medium"]).value
```

Comment:
**suggestion**: Given that the aggregations are returned from the API pre-sorted by popularity, we might be able to simplify this further by simply plucking out the first entry if present. It's also possible for this aggregation to be empty so we should guard for that as well.

```ts
const medium = getMediumSearchCriteriaLabel(
  getMostPopularOption(aggregations["medium"])
)

const getMostPopularOption = (aggregation: {
  [key: string]: { name: string; count: number }
}): string | undefined => {
  if (isExisty(aggregation)) {
    const key = Object.keys(aggregation)[0]
    return aggregation[key].value
  }
}
```

Reply:
good idea - do we always sort by popularity in all contexts?

Reply:
I believe so — yes, that's just how Elasticsearch aggregation are returned but it's up to the client whether to simply echo that sort order or do something different like Alphabetical sorting.